src/core/catalog.js

```diff
@@ -486,17 +486,17 @@ class Catalog {
         return shadow(this, "optionalContentConfig", null);
       }
       const groups = [];
-      const groupRefs = new RefSet();
+      const groupsMap = new Map();
```

Comment:
*This comment applies in multiple other spots below as well.*

Please note that it's not generally safe/correct to store `Ref`-instances in `Set`s and `Map`s, which is why we have `RefSet` respectively `RefSetCache` for that purpose; see [classes here](https://github.com/mozilla/pdf.js/blob/35a9a6a7b34030397c2215ebd3d17064ac53887e/src/core/primitives.js#L318-L391).

In this case you thus want the following instead.
```suggestion
      const groupRefCache = new RefSetCache();
```

Reply:
See the comment below.